src/picterra/client.py

```diff
@@ -82,22 +83,38 @@ def _wait_until_operation_completes(self, operation_response):
                 raise APIError('Operation %s failed' % operation_id)
             time.sleep(poll_interval)
 
-    def _paginate_through_list(self, resource_endpoint: str):
+    def _paginate_through_list(self, resource_endpoint: str, params: dict = {}):
+        # We explicit page number, even if it is not needed
         url = self._api_url('%s/?page_number=1' % resource_endpoint)
+        # Store list data, page (for debugging, future limit), total count (same as latter)
         data = []
+        page = 0
+        count = 0
+        # Loop over all the pages
         while url:
-            logger.debug('Paginating through %s list at page %s' % (resource_endpoint, url))
-            resp = self.sess.get(url)
+            page += 1
```

Comment:
I feel this whole thing could be rewritten into:

```
data = []
url = self._api_url('%s/?page_number=1' % resource_endpoint, params=params)
while url:
  logger.debug('fetching page url=%s', url)
  resp = self.sess.get(url)
  if not resp.ok:
    raise APIError(resp.text)
  r = resp.json()
  url = r['next']
  count = r['count']
  data += r['results']
assert len(data) == count, 'Error in pagination [%d!=%d]' % (len(data), count)
return data
```

No need for the page/count bookkeeping. It should be included in the logged url anyway.

Reply:
Checked (I put your version, just modified `_api_url` to add querystring params